driver/chain_syncer/calldata/syncer.go

```diff
@@ -80,19 +80,20 @@ func (s *Syncer) ProcessL1Blocks(ctx context.Context, l1End *types.Header) error
 		s.reorgDetectedFlag = false
 		firstTry = false
 
-		startHeight := s.state.GetL1Current().Number
+		startL1Current := s.state.GetL1Current()
 		// If there is a L1 reorg, sometimes this will happen.
-		if startHeight.Uint64() >= l1End.Number.Uint64() {
-			startHeight = new(big.Int).Sub(l1End.Number, common.Big1)
-			newL1Current, err := s.rpc.L1.HeaderByNumber(ctx, startHeight)
+		if startL1Current.Number.Uint64() >= l1End.Number.Uint64() && startL1Current.Hash() != l1End.Hash() {
+			newL1Current, err := s.rpc.L1.HeaderByNumber(ctx, new(big.Int).Sub(l1End.Number, common.Big1))
 			if err != nil {
 				return err
 			}
 
```

Comment:
do we not need to set `l1Current` to `newL1Current` instead of just using it in the log below? im not sure

Reply:
already on line 99 i think